docs/guides/Native_Integrations.md

```diff
@@ -0,0 +1,169 @@
+# Native Integrations
+
+By design, Elementary's native audio engine aims to easily fit into your audio
+processing stack to take on as much or as little of the processing responsibilities
+as you want. The engine can be embedded independently of any JavaScript
+processing, through a quick series of CMake and C++ steps which we
+outline here.
+
+## CMake
+
+To start, we add Elementary to your project with git submodules and CMake.
+
+```bash
+git submodule add https://github.com/elemaudio/elementary.git elementary
+```
+
+Then, in your `CMakeLists.txt`:
+
+```cmake
+# Add the runtime subdirectory
+add_subdirectory(elementary/runtime)
+
+# Make sure you link your target against the runtime
+target_link_libraries(${TARGET_NAME} PRIVATE runtime)
+```
+
+
+## C++
+
+Now that we've got the dependency in place, we can make an instance of the `elem::Runtime<FloatType>`. This Runtime
+instance is typically the only piece of the Elementary native engine that you need to pay any attention to, and using
+it properly only involves a handful of steps:
+
+1. Create the Runtime instance with your desired float type (`float` or `double`), sample rate, and block size
+2. Set up a message passing interface to receive instructions from your JavaScript environment and apply them to the runtime
+3. Process audio on the real-time thread (or wherever you need)
+4. (Optional) Process events periodically on the main thread
+
+As an example, here we'll show a simple C++ program which runs the Runtime instance next to an embedded JavaScript engine
+thanks to CHOC's Quickjs wrapper. We wire in a native interop method for receiving instructions from the JavaScript engine,
+then evaluate some JavaScript, and finally process a couple blocks of audio on the main thread.
+
+```cpp
+#include <Runtime.h>
+
+
+int main(int argc, char** argv) {
+    elem::Runtime<float> runtime(44100.0, 512);
+
+    auto ctx = choc::javascript::createQuickJSContext();
+
+    ctx.registerFunction("__postInstructions__", [&](choc::javascript::ArgumentList args) {
+        // Imagining here that we've JSON-stringified the instructions before calling into
+        // this native interop method, so we must parse the instruction set from JSON before applying.
+        runtime.applyInstructions(elem::js::parseJSON(args[0]->toString()));
+        return choc::value::Value();
+    });
+
+    // Now that we have the message passing set up, we can evaluate our JavaScript containing the
+    // Elementary frontend code
+    auto rv = ctx.evaluate(inputFileContents);
```

Comment:
Probably my lack of familiarity with C++, but how is `rv` used here? It doesn't appear to be used in the runtime processing.

Reply:
Good question; it's not. This is just the C++ API for evaluating an arbitrary expression in the javascript interpreter, which always resolves to _some_ return value (hence "rv"), even if that rv is often `undefined` when evaluating a big bundle or something